tests/python/relay/test_dataflow_pattern.py

```diff
@@ -1375,6 +1392,63 @@ def test_partition_overused():
     assert pattern.partition(out) == out
 
 
+def test_partition_fuzzy_tuple():
+    x = relay.var("x")
+    y = relay.var("y")
+    z = x + y
+    tuple_pattern = is_tuple(None)
```

Comment:
Why the use of None? What would happen if the brackets were left empty (similarly to is_constant())? (Just asking for enlightenment)

Reply:
I wanted to keep the API consistent between tuple, call, and function. There are many places were we may have functions or ops with no arguments, so calling an op with `op()` yields an empty input list, which is a concrete pattern we want to keep matching. I choose `None` to easily identify the fuzzy case against the empty input case.

Reply:
The problem is that we're not actually talking about a pattern, which explicitly represents an AST node. Instead, we're trying to represent an attribute of an AST node, particularly an attribute of the type Array<Expr>, that's what needs to be passed in. In the backend I need to be able to distinguish between a valid array of size 0 that I want to match on, and an invalid fuzzy array I don't want to match on. The simplest way to do that is to store a nullptr, and when I pass python None to the backend, that's what I get.

I could introduce something like an `arg+` pattern that goes into the argument list, but that screws with the locality of the pattern matcher and creates a need to do a lot of look-ahead to determine how to match arguments so I don't accidentally index into an array I shouldn't or reject the match because the arguments don't match. It's feasible, but messy.

Reply:
@ekalda @mbaret do we agree with the interface?

Reply:
I agree with @ekalda, in the absence of a more self-documenting interface explicit documentation is the next best option.

Reply:
This makes sense, thanks! I'll add this to the docs (fix my spelling mistake :D) And re-push.